diff --git a/Code/RDGeneral/RDValue-taggedunion.h b/Code/RDGeneral/RDValue-taggedunion.h
index 3efb2bcdd..fd6d99c94 100644
--- a/Code/RDGeneral/RDValue-taggedunion.h
+++ b/Code/RDGeneral/RDValue-taggedunion.h
@@ -39,7 +39,6 @@
 #include <vector>
 #include <cstdint>
 #include <RDGeneral/BoostStartInclude.h>
-#include <cstdint>
 #include <any>
 #include <boost/utility.hpp>
 #include <boost/lexical_cast.hpp>
