diff --git a/rdkit/Chem/Pharm2D/LazyGenerator.py b/rdkit/Chem/Pharm2D/LazyGenerator.py
index 8e8c53938..5e72869c7 100755
--- a/rdkit/Chem/Pharm2D/LazyGenerator.py
+++ b/rdkit/Chem/Pharm2D/LazyGenerator.py
@@ -77,10 +77,7 @@ class Generator(object):
     for feat in feats:
       if feat.GetFamily() not in sigFactory.skipFeats:
         featMatches[feat.GetFamily()].append(feat.GetAtomIds())
-    featMatches = [None] * nFeats
-    for i in range(nFeats):
-      featMatches[i] = sigFactory.featFactory.GetMolFeature()
-    self.pattMatches = pattMatches
+    self.pattMatches = [featMatches[family] for family in featFamilies]
 
   def GetBit(self, idx):
     """ returns a bool indicating whether or not the bit is set
