diff --git a/Code/GraphMol/SmilesParse/SmartsWrite.cpp b/Code/GraphMol/SmilesParse/SmartsWrite.cpp
--- a/Code/GraphMol/SmilesParse/SmartsWrite.cpp
+++ b/Code/GraphMol/SmilesParse/SmartsWrite.cpp
@@ -463,7 +463,7 @@ std::string _recurseBondSmarts(
   } else {
     std::stringstream msg;
     msg << "Can't write smarts for this query bond type: " << descrip;
-    throw msg.str().c_str();
+    throw ValueErrorException(msg.str());
   }
   return res;
 }
