diff --git a/Code/GraphMol/ChemTransforms/MolZip.cpp b/Code/GraphMol/ChemTransforms/MolZip.cpp
--- a/Code/GraphMol/ChemTransforms/MolZip.cpp
+++ b/Code/GraphMol/ChemTransforms/MolZip.cpp
@@ -715,7 +715,7 @@ std::unique_ptr<ROMol> molzip(std::vector<ROMOL_SPTR> &decomposition,
     const auto zippedConf = zippedMol->getConformer(confId);
     auto attachmentMapping = *attachmentMappingOption;
     for (auto &mol : decomposition) {
-      const auto newConf = new Conformer(mol->getNumAtoms());
+      auto newConf = std::make_unique<Conformer>(mol->getNumAtoms());
       newConf->set3D(false);
       for (const auto atom : mol->atoms()) {
         int zippedIndex = atom->getProp<int>(indexPropName);
@@ -731,11 +731,13 @@ std::unique_ptr<ROMol> molzip(std::vector<ROMOL_SPTR> &decomposition,
               const auto index = zippedAtom->getProp<int>(indexPropName);
               return index == zippedIndex;
             });
+        CHECK_INVARIANT(zippedAtom != zipppedAtoms.end(),
+                        "molzip: cannot map atom into zipped molecule");
 
         newConf->setAtomPos(atom->getIdx(),
                             zippedConf.getAtomPos((*zippedAtom)->getIdx()));
       }
-      mol->addConformer(newConf, true);
+      mol->addConformer(newConf.release(), true);
     }
     for (const auto atom : zippedMol->atoms()) {
       atom->clearProp(indexPropName);
