diff --git a/rdkit/Chem/BuildFragmentCatalog.py b/rdkit/Chem/BuildFragmentCatalog.py
--- a/rdkit/Chem/BuildFragmentCatalog.py
+++ b/rdkit/Chem/BuildFragmentCatalog.py
@@ -580,5 +580,5 @@ if __name__ == '__main__':
       except Exception:
         obls = None
       else:
-        if len(obls) < (inD.count('\n') - 1):
+        if details.numMols > 0 and len(obls) < details.numMols:
           obls = None
