diff --git a/Code/GraphMol/MolStandardize/Metal.cpp b/Code/GraphMol/MolStandardize/Metal.cpp
index f98b712d1..202410ae9 100644
--- a/Code/GraphMol/MolStandardize/Metal.cpp
+++ b/Code/GraphMol/MolStandardize/Metal.cpp
@@ -15,7 +15,6 @@
 #include <GraphMol/Substruct/SubstructMatch.h>
 
 #include <GraphMol/RDKitQueries.h>
-#include <GraphMol/Substruct/SubstructMatch.h>
 #include <GraphMol/Substruct/SubstructUtils.h>
 
 using namespace std;
