diff --git a/Code/GraphMol/ROMol.h b/Code/GraphMol/ROMol.h
index 48a798fe7..30c8fc649 100644
--- a/Code/GraphMol/ROMol.h
+++ b/Code/GraphMol/ROMol.h
@@ -515,7 +515,7 @@ class RDKIT_GRAPHMOL_EXPORT ROMol : public RDProps {
   //! construct a molecule from a pickle string
   ROMol(const std::string &binStr, unsigned int propertyFlags);
 
-  ROMol(ROMol &&o) noexcept
+  ROMol(ROMol &&o)
       : RDProps(std::move(o)),
         d_graph(std::move(o.d_graph)),
         d_atomBookmarks(std::move(o.d_atomBookmarks)),
@@ -542,7 +542,7 @@ class RDKIT_GRAPHMOL_EXPORT ROMol : public RDProps {
     dp_delAtoms = std::exchange(o.dp_delAtoms, nullptr);
     dp_delBonds = std::exchange(o.dp_delBonds, nullptr);
   }
-  ROMol &operator=(ROMol &&o) noexcept {
+  ROMol &operator=(ROMol &&o) {
     if (this == &o) {
       return *this;
     }
