diff --git a/Code/SimDivPickers/HierarchicalClusterPicker.cpp b/Code/SimDivPickers/HierarchicalClusterPicker.cpp
index 7da6a8897..541301fb7 100644
--- a/Code/SimDivPickers/HierarchicalClusterPicker.cpp
+++ b/Code/SimDivPickers/HierarchicalClusterPicker.cpp
@@ -27,21 +27,18 @@ RDKit::VECT_INT_VECT HierarchicalClusterPicker::cluster(
   // Do the clustering
   auto method = (long int)d_method;
   long int len = poolSize * (poolSize - 1);
-  auto *ia = (long int *)calloc(poolSize, sizeof(long int));
-  auto *ib = (long int *)calloc(poolSize, sizeof(long int));
-  real *crit = (real *)calloc(poolSize, sizeof(real));
-  CHECK_INVARIANT(ia, "failed to allocate memory");
-  CHECK_INVARIANT(ib, "failed to allocate memory");
-  CHECK_INVARIANT(crit, "failed to allocate memory");
+  std::vector<long int> ia(poolSize);
+  std::vector<long int> ib(poolSize);
+  std::vector<real> crit(poolSize);
   auto poolSize2 = static_cast<long int>(poolSize);
 
   distdriver_(&poolSize2,       // number of items in the pool
               &len,             // number of entries in the distance matrix
               (real *)distMat,  // distance matrix
               &method,          // the clustering method (ward, slink etc.)
-              ia,               // int vector with clustering history
-              ib,               // one more clustering history matrix
-              crit  // I believe this is a vector the difference in heights of
+              ia.data(),        // int vector with clustering history
+              ib.data(),        // one more clustering history matrix
+              crit.data()  // I believe this is a vector the difference in heights of
                     // two clusters
   );
 
@@ -76,10 +73,6 @@ RDKit::VECT_INT_VECT HierarchicalClusterPicker::cluster(
     // mark the second cluster as removed
     removed.push_back(cx2);
   }
-  free(ia);
-  free(ib);
-  free(crit);
-
   // sort removed so that looping will be easier later
   std::sort(removed.begin(), removed.end());
 
