diff --git a/Code/GraphMol/RascalMCES/RascalMCES.cpp b/Code/GraphMol/RascalMCES/RascalMCES.cpp
--- a/Code/GraphMol/RascalMCES/RascalMCES.cpp
+++ b/Code/GraphMol/RascalMCES/RascalMCES.cpp
@@ -822,7 +822,8 @@ void checkTimeout(
         if (clique.size() > maxCliques.front().size()) {
           maxCliques.clear();
         }
-        if (clique.size() >= maxCliques.front().size()) {
+        if (maxCliques.empty() ||
+            clique.size() >= maxCliques.front().size()) {
           maxCliques.push_back(clique);
         }
       }
