diff --git a/Code/GraphMol/Canon.cpp b/Code/GraphMol/Canon.cpp
index c29de154b..3f1c51ec0 100644
--- a/Code/GraphMol/Canon.cpp
+++ b/Code/GraphMol/Canon.cpp
@@ -184,9 +184,9 @@ bool fixConflictAcrossDoubleBond(const Bond &dblBond, const Atom &atom,
 bool handleDirConflictsAcrossDoubleBond(
     const Bond &dblBond, const Atom &atom1, bool atom1DirsAreConsistent,
     const Bond &firstFromAtom1, bool isFirstFromAtom1Flipped,
-    const Bond &secondFromAtom1, bool isSecondFromAtom1Flipped,
+    const Bond *secondFromAtom1, bool isSecondFromAtom1Flipped,
     const Atom &atom2, bool atom2DirsAreConsistent, const Bond &firstFromAtom2,
-    bool isFirstFromAtom2Flipped, const Bond &secondFromAtom2,
+    bool isFirstFromAtom2Flipped, const Bond *secondFromAtom2,
     bool isSecondFromAtom2Flipped, std::vector<int8_t> &bondDirCounts,
     std::vector<int8_t> &atomDirCounts
 
@@ -208,7 +208,7 @@ bool handleDirConflictsAcrossDoubleBond(
     // we know it is not problematic.
     return fixConflictAcrossDoubleBond(
         dblBond, atom2, firstFromAtom2, isFirstFromAtom2Flipped,
-        secondFromAtom2, isSecondFromAtom2Flipped, atom1, firstFromAtom1,
+        *secondFromAtom2, isSecondFromAtom2Flipped, atom1, firstFromAtom1,
         isFirstFromAtom1Flipped, bondDirCounts, atomDirCounts);
 
   } else if (!atom1DirsAreConsistent && atom2DirsAreConsistent) {
@@ -218,7 +218,7 @@ bool handleDirConflictsAcrossDoubleBond(
     // we know it is not problematic.
     return fixConflictAcrossDoubleBond(
         dblBond, atom1, firstFromAtom1, isFirstFromAtom1Flipped,
-        secondFromAtom1, isSecondFromAtom1Flipped, atom2, firstFromAtom2,
+        *secondFromAtom1, isSecondFromAtom1Flipped, atom2, firstFromAtom2,
         isFirstFromAtom2Flipped, bondDirCounts, atomDirCounts);
 
   } else {
@@ -228,32 +228,32 @@ bool handleDirConflictsAcrossDoubleBond(
     // to be removed to end up with a valid across-double-bond configuration.
 
     for (const auto &[atom1Bond, atom1BondisFlipped] :
-         {std::make_pair(firstFromAtom1, isFirstFromAtom1Flipped),
+         {std::make_pair(&firstFromAtom1, isFirstFromAtom1Flipped),
           std::make_pair(secondFromAtom1, isSecondFromAtom1Flipped)}) {
       for (const auto &[atom2Bond, atom2BondisFlipped] :
-           {std::make_pair(firstFromAtom2, isFirstFromAtom2Flipped),
+           {std::make_pair(&firstFromAtom2, isFirstFromAtom2Flipped),
             std::make_pair(secondFromAtom2, isSecondFromAtom2Flipped)}) {
         auto expectedAtom2Dir = getReferenceDirection(
-            dblBond, atom1, atom2, atom1Bond, atom1BondisFlipped, atom2Bond,
+            dblBond, atom1, atom2, *atom1Bond, atom1BondisFlipped, *atom2Bond,
             atom2BondisFlipped);
-        if (expectedAtom2Dir == atom2Bond.getBondDir()) {
+        if (expectedAtom2Dir == atom2Bond->getBondDir()) {
           // We have found a combination of directions that are consistent with
           // the double bond's stereo label. Now we need to check if we can
           // remove the other two directions to fix the conflict.
 
           auto atom1OtherBond =
-              (&atom1Bond == &firstFromAtom1 ? secondFromAtom1
-                                             : firstFromAtom1);
-          auto atom1OtherIdx = atom1OtherBond.getOtherAtomIdx(atom1.getIdx());
+              (atom1Bond == &firstFromAtom1 ? secondFromAtom1
+                                           : &firstFromAtom1);
+          auto atom1OtherIdx = atom1OtherBond->getOtherAtomIdx(atom1.getIdx());
           auto canAtom1OtherDirBeRemoved = atomDirCounts[atom1OtherIdx] == 2;
           if (!canAtom1OtherDirBeRemoved) {
             continue;
           }
 
           auto atom2OtherBond =
-              (&atom2Bond == &firstFromAtom2 ? secondFromAtom2
-                                             : firstFromAtom2);
-          auto atom2OtherIdx = atom2OtherBond.getOtherAtomIdx(atom2.getIdx());
+              (atom2Bond == &firstFromAtom2 ? secondFromAtom2
+                                           : &firstFromAtom2);
+          auto atom2OtherIdx = atom2OtherBond->getOtherAtomIdx(atom2.getIdx());
           if (atom1OtherIdx == atom2OtherIdx) {
             // unlikely, but not impossible, so just in case...
             continue;
@@ -264,11 +264,11 @@ bool handleDirConflictsAcrossDoubleBond(
             continue;
           }
 
-          bondDirCounts[atom1OtherBond.getIdx()] = 0;
+          bondDirCounts[atom1OtherBond->getIdx()] = 0;
           --atomDirCounts[atom1.getIdx()];
           --atomDirCounts[atom1OtherIdx];
 
-          bondDirCounts[atom2OtherBond.getIdx()] = 0;
+          bondDirCounts[atom2OtherBond->getIdx()] = 0;
           --atomDirCounts[atom2.getIdx()];
           --atomDirCounts[atom2OtherIdx];
 
@@ -546,9 +546,9 @@ void canonicalizeDoubleBond(Bond *dblBond, const UINT_VECT &bondVisitOrders,
     // with what we see on each end of the double bond
     if (!handleDirConflictsAcrossDoubleBond(
             *dblBond, *atom1, atom1DirsAreConsistent, *firstFromAtom1,
-            isFirstFromAtom1Flipped, *secondFromAtom1, isSecondFromAtom1Flipped,
+            isFirstFromAtom1Flipped, secondFromAtom1, isSecondFromAtom1Flipped,
             *atom2, atom2DirsAreConsistent, *firstFromAtom2,
-            isFirstFromAtom2Flipped, *secondFromAtom2, isSecondFromAtom2Flipped,
+            isFirstFromAtom2Flipped, secondFromAtom2, isSecondFromAtom2Flipped,
             bondDirCounts, atomDirCounts)) {
       logInconsistentBondDirsWarning(dblBond->getIdx());
     }
