diff --git a/Code/GraphMol/FileParsers/MolFileWriter.cpp b/Code/GraphMol/FileParsers/MolFileWriter.cpp
--- a/Code/GraphMol/FileParsers/MolFileWriter.cpp
+++ b/Code/GraphMol/FileParsers/MolFileWriter.cpp
@@ -634,6 +634,6 @@ std::string getV2000AtomBlock(const ROMol &mol, const Atom *atom,
 #ifndef _MSC_VER
   snprintf(dest, 128,
-           "%10.4f%10.4f%10.4f %3s%2d%3d%3d%3d%3d%3d  0%3d%3d%3d%3d%3d", x, y,
+           "%10.4f%10.4f%10.4f %3s%2d%3d%3u%3d%3d%3d  0%3d%3d%3d%3d%3d", x, y,
            z, symbol.c_str(), massDiff, chg, parityFlag, hCount, stereoCare,
            totValence, rxnComponentType, rxnComponentNumber, atomMapNumber,
            inversionFlag, exactChangeFlag);
@@ -645,6 +645,6 @@ std::string getV2000AtomBlock(const ROMol &mol, const Atom *atom,
   // about use of sprintf
   sprintf_s(dest, 128,
-            "%10.4f%10.4f%10.4f %3s%2d%3d%3d%3d%3d%3d  0%3d%3d%3d%3d%3d", x, y,
+            "%10.4f%10.4f%10.4f %3s%2d%3d%3u%3d%3d%3d  0%3d%3d%3d%3d%3d", x, y,
             z, symbol.c_str(), massDiff, chg, parityFlag, hCount, stereoCare,
             totValence, rxnComponentType, rxnComponentNumber, atomMapNumber,
             inversionFlag, exactChangeFlag);
