diff --git a/Code/GraphMol/FileParsers/MaeWriter.cpp b/Code/GraphMol/FileParsers/MaeWriter.cpp
--- a/Code/GraphMol/FileParsers/MaeWriter.cpp
+++ b/Code/GraphMol/FileParsers/MaeWriter.cpp
@@ -542,8 +542,8 @@ MaeWriter::MaeWriter(std::ostream *outStream) : dp_ostream{outStream} {
 
 MaeWriter::MaeWriter(std::shared_ptr<std::ostream> outStream)
     : dp_ostream{std::move(outStream)} {
-  PRECONDITION(outStream, "null stream");
-  if (outStream->bad()) {
+  PRECONDITION(dp_ostream, "null stream");
+  if (dp_ostream->bad()) {
     throw FileParseException("Bad output stream");
   }
 }
