diff --git a/rdkit/ML/Data/DataUtils.py b/rdkit/ML/Data/DataUtils.py
index 93035d12a..379765175 100755
--- a/rdkit/ML/Data/DataUtils.py
+++ b/rdkit/ML/Data/DataUtils.py
@@ -655,4 +655,4 @@ def RandomizeActivities(dataSet, shuffle=0, runDetails=None):
-  else:  # This part of the code isn't working as examples is not defined
+  else:
     if runDetails:
       runDetails.randomized = 1
     nPossible = dataSet.GetNPossibleVals()[-1]
