diff --git a/Code/GraphMol/ShapeHelpers/ShapeEncoder.cpp b/Code/GraphMol/ShapeHelpers/ShapeEncoder.cpp
index c2d141d0d..f59d4ab30 100644
--- a/Code/GraphMol/ShapeHelpers/ShapeEncoder.cpp
+++ b/Code/GraphMol/ShapeHelpers/ShapeEncoder.cpp
@@ -11,9 +11,7 @@
 #include <Geometry/UniformGrid3D.h>
 
 #include "ShapeEncoder.h"
-#include <Geometry/Transform3D.h>
 #include <Geometry/point.h>
-#include <Geometry/UniformGrid3D.h>
 #include <GraphMol/Conformer.h>
 #include <GraphMol/RDKitBase.h>
 namespace RDKit {
