diff --git a/Code/GraphMol/MolDraw2D/AtomSymbol.cpp b/Code/GraphMol/MolDraw2D/AtomSymbol.cpp
--- a/Code/GraphMol/MolDraw2D/AtomSymbol.cpp
+++ b/Code/GraphMol/MolDraw2D/AtomSymbol.cpp
@@ -135,9 +135,10 @@ void AtomSymbol::adjustColons() {
   if (colonPos == std::string::npos) {
     return;
   }
-  CHECK_INVARIANT(colonPos <= rects_.size(), "bad rects_ size");
+  CHECK_INVARIANT(colonPos < rects_.size(), "bad rects_ size");
   double leftHeight = colonPos ? rects_[colonPos - 1]->height_ : 0;
-  double rightHeight =
-      colonPos < symbol_.size() - 1 ? rects_[colonPos + 1]->height_ : 0;
+  double rightHeight = colonPos + 1 < rects_.size()
+                           ? rects_[colonPos + 1]->height_
+                           : 0;
   rects_[colonPos]->height_ = std::min(leftHeight, rightHeight);
 }
