diff --git a/Code/GraphMol/MolDraw2D/DrawShape.cpp b/Code/GraphMol/MolDraw2D/DrawShape.cpp
index c8763f3f8..fdedbb681 100644
--- a/Code/GraphMol/MolDraw2D/DrawShape.cpp
+++ b/Code/GraphMol/MolDraw2D/DrawShape.cpp
@@ -314,7 +314,7 @@ bool DrawShapePolyLine::doesRectClash(const StringRect &rect,
 }
 
 // ****************************************************************************
-DrawShapeSolidWedge::DrawShapeSolidWedge(const std::vector<Point2D> points,
+DrawShapeSolidWedge::DrawShapeSolidWedge(const std::vector<Point2D> &points,
                                          const DrawColour &col1,
                                          const DrawColour &col2,
                                          bool splitBonds,
@@ -550,7 +550,7 @@ void DrawShapeSolidWedge::orderOtherBondVecs() {
 }
 
 // ****************************************************************************
-DrawShapeDashedWedge::DrawShapeDashedWedge(const std::vector<Point2D> points,
+DrawShapeDashedWedge::DrawShapeDashedWedge(const std::vector<Point2D> &points,
                                            const DrawColour &col1,
                                            const DrawColour &col2,
                                            bool oneLessDash, double lineWidth,
@@ -679,7 +679,7 @@ bool DrawShapeDashedWedge::doesRectClash(const StringRect &rect,
 }
 
 // ****************************************************************************
-DrawShapeWavyLine::DrawShapeWavyLine(const std::vector<Point2D> points,
+DrawShapeWavyLine::DrawShapeWavyLine(const std::vector<Point2D> &points,
                                      double lineWidth, bool scaleLineWidth,
                                      const DrawColour &col1,
                                      const DrawColour &col2, double offset,
@@ -717,7 +717,7 @@ bool DrawShapeWavyLine::doesRectClash(const StringRect &rect,
 }
 
 // ****************************************************************************
-DrawShapeArc::DrawShapeArc(const std::vector<Point2D> points, double ang1,
+DrawShapeArc::DrawShapeArc(const std::vector<Point2D> &points, double ang1,
                            double ang2, double lineWidth, bool scaleLineWidth,
                            const DrawColour &col1, bool fill, int atom1)
     : DrawShape(points, lineWidth, scaleLineWidth, col1, fill, atom1),
diff --git a/Code/GraphMol/MolDraw2D/DrawShape.h b/Code/GraphMol/MolDraw2D/DrawShape.h
index 895dc5225..a84bb2e91 100644
--- a/Code/GraphMol/MolDraw2D/DrawShape.h
+++ b/Code/GraphMol/MolDraw2D/DrawShape.h
@@ -142,7 +142,7 @@ class DrawShapePolyLine : public DrawShape {
 
 class DrawShapeSolidWedge : public DrawShape {
  public:
-  DrawShapeSolidWedge(const std::vector<Point2D> points, const DrawColour &col1,
+  DrawShapeSolidWedge(const std::vector<Point2D> &points, const DrawColour &col1,
                       const DrawColour &col2, bool splitBonds,
                       std::vector<Point2D> &otherBondVecs,
                       double lineWidth = 1.0, int atom1 = -1, int atom2 = -1,
@@ -177,7 +177,7 @@ class DrawShapeDashedWedge : public DrawShape {
   // isn't drawn.  The wedge will be as fat as it would have been with
   // the extra dash.  This is so that bonds coming out of the fat end
   // of the wedge aren't directly incident on a dash.
-  DrawShapeDashedWedge(const std::vector<Point2D> points,
+  DrawShapeDashedWedge(const std::vector<Point2D> &points,
                        const DrawColour &col1, const DrawColour &col2,
                        bool oneLessDash = true, double lineWidth = 1.0,
                        int atom1 = -1, int atom2 = -1, int bond = -1);
@@ -204,7 +204,7 @@ class DrawShapeDashedWedge : public DrawShape {
 
 class DrawShapeWavyLine : public DrawShape {
  public:
-  DrawShapeWavyLine(const std::vector<Point2D> points, double lineWidth = 2.0,
+  DrawShapeWavyLine(const std::vector<Point2D> &points, double lineWidth = 2.0,
                     bool scaleLineWidth = false,
                     const DrawColour &col1 = DrawColour(0, 0, 0),
                     const DrawColour &col2 = DrawColour(0, 0, 0),
@@ -233,7 +233,7 @@ class DrawShapeArc : public DrawShape {
   // between 0 and 360.0.
   // Points should be size 2 - the first entry is the centre, the second
   // gives the x and y radii of the ellipse.
-  DrawShapeArc(const std::vector<Point2D> points, double ang1, double ang2,
+  DrawShapeArc(const std::vector<Point2D> &points, double ang1, double ang2,
                double lineWidth = 2.0, bool scaleLineWidth = false,
                const DrawColour &col1 = DrawColour(0, 0, 0), bool fill = false,
                int atom1 = -1);
